### PR TITLE
Update callers to Nice Partials method accessors

### PR DESCRIPTION
**Pull request by kaspth**
Also renames certain partial variables to make them more contextual or easier to spot the rendering when the partials are long. The pattern is to use the last segment for the variable name, e.g. `account/shared/page` and `account/shared/box` gets page and box variables respectively.

### CODE REVIEW

File: bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-      <% p.content_for :title, t('.header') %>
-      <% p.content_for :description, t('.description') %>
-      <% p.content_for :body do %>
-        <%= render 'form', access_token: @access_token %>
-      <% end %>
-    <% end %>
+<%= render 'account/shared/page' do |page| %>
+  <% page.title t('.section') %>
+  <% page.body.render 'account/shared/box', divider: true do |box| %>
+    <% box.t :description, title: '.header' %>
```
Comment (kaspth) [inline bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb:4]: Uses the `partial.t` helper to condense the translate call.

File: bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb [hunk 2]
```diff
@@ -1,12 +1,7 @@
...
-    <%= render 'account/shared/box', divider: true do |p| %>
-      <% p.content_for :title, t('.header') %>
-      <% p.content_for :description, t('.description') %>
-      <% p.content_for :body do %>
-        <%= render 'form', access_token: @access_token %>
-      <% end %>
-    <% end %>
+<%= render 'account/shared/page' do |page| %>
+  <% page.title t('.section') %>
+  <% page.body.render 'account/shared/box', divider: true do |box| %>
```
Comment (kaspth) [inline bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb:3]: Uses the forwarder on `partial.x` to automatically capture the `render` call and sparing us the extra block indentation.

File: bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb [hunk 3]
```diff
@@ -1,12 +1,7 @@
...
-  <% p.content_for :body do %>
-    <%= render 'account/shared/box', divider: true do |p| %>
-      <% p.content_for :title, t('.header') %>
-      <% p.content_for :description, t('.description') %>
-      <% p.content_for :body do %>
-        <%= render 'form', access_token: @access_token %>
-      <% end %>
-    <% end %>
+<%= render 'account/shared/page' do |page| %>
+  <% page.title t('.section') %>
```
Comment (kaspth) [inline bullet_train-api/app/views/account/platform/access_tokens/edit.html.erb:2]: Debated with whether or not to use the content passing here:

```suggestion
<%= render 'account/shared/page', title: t('.section') do |page| %>
```

…but ultimately went with the old version just so the callsites are a little more consistent.

**Review by andrewculver — APPROVED** [review]

**Comment by andrewculver** [comment]
@kaspth Holy cow dude. Feels like months of effort finally coming together and the result is really beautiful! Feels like a massive DX improvement for Nice Partials and Bullet Train downstream of that. Thank you for the vision on this one!

**Pull request merged**